app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -11,7 +11,7 @@ def initialize(authentication_parameters)
           @logger = Rails.logger
         end
 
-        def provide_jwt_identity
+        def jwt_identity
```

Comment:
you can change the method to:
```
        def jwt_identity
          return @jwt_identity if @jwt_identity

          token_field_name = fetch_token_field_name          @logger.debug(LogMessages::Authentication::AuthnJwt::CHECKING_IDENTITY_FIELD_EXISTS.new(token_field_name))          @jwt_identity = @decoded_token[token_field_name]          if @jwt_identity.blank?            raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken, token_field_name          end
          @logger.debug(LogMessages::Authentication::AuthnJwt::FOUND_JWT_FIELD_IN_TOKEN.new(token_field_name, @jwt_identity))          @jwt_identity        end
```

to memoize it

Reply:
ValidateResourceRestrictions recivies the validator as authentication_request
https://github.com/cyberark/conjur/blob/ba3a03e841eebe427b340bda5cf5983be1029f6e/app/domain/authentication/resource_restrictions/validate_resource_restrictions.rb#L28

I also don't agree with the naming but don't think its in scope for the jwt authenticator to fix this for all authenticators.

Reply:
Done